docs/appendices/release-notes/unreleased.rst

```diff
@@ -77,6 +77,9 @@ Fixes
 .. stable branch. You can add a version label (`v/X.Y`) to the pull request for
 .. an automated mergify backport.
 
+- Fixed an issue that lead multiple nested joins to run into an internal error.
+  As part of this the pre-validation of join ``using`` was also improved.
```

Comment:
Maybe add a concrete example.  Otherwise it's difficult to understand what exactly was fixed

Reply:
:+1: